### PR TITLE
Fixes #7529 - Add CRUD tests for http-proxy component.

### LINKED ISSUE #7529: Add CRUD test for http-proxy hammer command.
Hammer support for http-proxy[1] is now available in Satellite 6.7
Add CRUD test for http-proxy via hammer.
[1] [https://bugzilla.redhat.com/show_bug.cgi?id=1774325](https://bugzilla.redhat.com/show_bug.cgi?id=1774325)

### PR DESCRIPTION
**Pull request by jameerpathan111**
Fixes #7529

### CODE REVIEW

File: robottelo/cli/factory.py [hunk 1]
```diff
@@ -4316,3 +4317,71 @@ def virt_who_hypervisor_config(
...
+        'location-titles': None,
+        'name': gen_string('alpha', 15),
+        'organization': None,
+        'organization-id': None,
+        'organization-title': None,
+        'organizations': None,
+        'organization-ids': None,
+        'organization-titles': None,
+        'password': None,
+        'url': 'https://{}'.format(gen_string('alpha', 15)),
```
Comment (vijay8451) [inline robottelo/cli/factory.py]:
```suggestion
        'url': '{}:{}'.format(gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999)),
```
1. https url could genrate using `gen_url`
2. Add the port as well with URL.

File: tests/foreman/cli/test_http_proxy.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    tier2,
+    upgrade
+)
+from robottelo.test import CLITestCase
+
+
+class HttpProxyTestCase(CLITestCase):
+    """Tests for http-proxy via Hammer CLI"""
+
+    @tier2
```
Comment (vijay8451) [inline tests/foreman/cli/test_http_proxy.py]:
```suggestion
    @tier1
```

File: tests/foreman/cli/test_http_proxy.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+"""Tests for http-proxy hammer command.
+
+:Requirement: HttpProxy
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Repositories
```
Comment (vijay8451) [inline tests/foreman/cli/test_http_proxy.py]: I think component would be different but it's pending on me and I will update you on it.

File: tests/foreman/cli/test_http_proxy.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        :CaseImportance: Critical
+        """
+        loc = make_location()
+        org = make_org()
+        self.addCleanup(location_cleanup, loc['id'])
+        self.addCleanup(org_cleanup, org['id'])
+        # Create http proxy
+        name = gen_string('alpha', 15)
+        url = 'https://{}'.format(gen_string('alpha', 15))
```
Comment (vijay8451) [inline tests/foreman/cli/test_http_proxy.py]:
```suggestion
        url = '{}:{}'.format(gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999))
```

File: tests/foreman/cli/test_http_proxy.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        password = gen_string('alpha', 15)
+        username = gen_string('alpha', 15)
+        updated_name = gen_string('alpha', 15)
+        updated_url = 'https://{}'.format(gen_string('alpha', 15))
```
Comment (vijay8451) [inline tests/foreman/cli/test_http_proxy.py]: same as above.

**Comment by vijay8451** [comment]
Share the latest tests results as well.

**Comment by jameerpathan111** [comment]
Test results:
```
$ pytest tests/foreman/cli/test_http_proxy.py 
=================================================================================== test session starts ===================================================================================
collected 1 item                                                                                                                                                                          

tests/foreman/cli/test_http_proxy.py .                                                                                                                                              [100%]

=============================================================================== 1 passed in 119.19 seconds ================================================================================
```

```
$ pytest tests/foreman/ui/test_http_proxy.py 
=================================================================================== test session starts ===================================================================================
collected 1 item                                                                                                                                                                          

tests/foreman/ui/test_http_proxy.py .                                                                                                                                               [100%]

==================================================================================== warnings summary =====================================================================================
tests/foreman/ui/test_http_proxy.py::test_positive_end_to_end
tests/foreman/ui/test_http_proxy.py::test_positive_end_to_end
  /home/jpathan/projects/env-robottelo/lib64/python3.6/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================================= 1 passed, 2 warnings in 125.86 seconds ==========================================================================
```

File: tests/foreman/cli/test_http_proxy.py (hunk 5 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_http_proxy.py]: updated PR.

File: tests/foreman/cli/test_http_proxy.py (hunk 4 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_http_proxy.py]: updated PR.

File: tests/foreman/cli/test_http_proxy.py (hunk 3 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_http_proxy.py]: ok

File: tests/foreman/cli/test_http_proxy.py (hunk 2 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_http_proxy.py]: updated PR.

File: robottelo/cli/factory.py (hunk 1 above)
Comment (jameerpathan111) [reply robottelo/cli/factory.py]: Thanks, updated PR.

**Comment by jameerpathan111** [comment]
Airgun PR https://github.com/SatelliteQE/airgun/pull/432 needs to be merged as well.

File: tests/foreman/ui/test_http_proxy.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+:Upstream: No
+"""
+from fauxfactory import gen_string, gen_url, gen_integer
+from robottelo.decorators import tier1, upgrade
+
+
+@tier1
+@upgrade
+def test_positive_end_to_end(session, module_org, module_loc):
```
Comment (vijay8451) [inline tests/foreman/ui/test_http_proxy.py]:
```suggestion
def test_positive_create_update_delete(session, module_org, module_loc):
```

File: tests/foreman/ui/test_http_proxy.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+from fauxfactory import gen_string, gen_url, gen_integer
+from robottelo.decorators import tier1, upgrade
+
+
+@tier1
```
Comment (vijay8451) [inline tests/foreman/ui/test_http_proxy.py]:
```suggestion
@tier2
```

File: tests/foreman/ui/test_http_proxy.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+@upgrade
+def test_positive_end_to_end(session, module_org, module_loc):
+    """Perform end to end testing for http-proxy component.
```
Comment (vijay8451) [inline tests/foreman/ui/test_http_proxy.py]:
```suggestion
    """Create new http-proxy with attributes, update and delete it.
```

Comment (jameerpathan111) [reply tests/foreman/ui/test_http_proxy.py]: Updated PR.

File: tests/foreman/ui/test_http_proxy.py (hunk 7 above)
Comment (jameerpathan111) [reply tests/foreman/ui/test_http_proxy.py]: Updated PR.

File: tests/foreman/ui/test_http_proxy.py (hunk 6 above)
Comment (jameerpathan111) [reply tests/foreman/ui/test_http_proxy.py]: Updated PR.

**Pull request merged**